### PR TITLE
feat: gh-actions for release & pr; support macos, tvos, watchos

### PR DESCRIPTION
**Pull request by bgiori**
This PR adds quite a bit:

**Automation**
1. Release process automation using [semantic-release](https://semantic-release.gitbook.io/semantic-release/) and github actions.
    - PRs must now follow the [semantic rules](https://github.com/amplitude/Amplitude-iOS/blob/main/CONTRIBUTING.md) 
2. PR Checks via github actions for linting, building, and testing.
3. PR template based on Amplitude-iOS

**Platform Support**
* Support for macOS, tvOS, and watchOS
* Platform specific tests in release process and PRs

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        run: |
+          git clone https://github.com/SwiftDocOrg/swift-doc
+          cd swift-doc
+          git checkout ${{ env.SWIFT_DOC_VERSION }}
+          make install
+          cd ..
+          rm -rf swift-doc
+
+      - name: Generate Swift Doc
+        run: swift doc generate Sources/Skylab/ --module-name Skylab --output docs --format html --base-url /skylab-ios-client
```
Comment (curtisliu) [inline .github/workflows/release.yml:92]: `skylab-ios-client` can probably be pulled into a variable that is easy to change based on the repo name

**Review by curtisliu — APPROVED** [review]
For full support of macOS/tvOS/watch we'd want to test to make sure values provided in https://github.com/amplitude/skylab-ios-client/blob/main/Sources/Skylab/AmplitudeContextProvider.swift are correct.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: Lint
+
+on: [pull_request]
```
Comment (jooohhn) [inline .github/workflows/lint.yml:3]: I think we'll want to use `[push, pull_request]` or else `lint.yml` won't run for commits pushed after a PR is created

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Test
+
+on: [pull_request]
```
Comment (jooohhn) [inline .github/workflows/test.yml:3]: I think we want to use `[push, pull_request]` or else tests won't run for pushes created after a PR is created and when a PR is merged into main branch

**Review by jooohhn — CHANGES_REQUESTED** [review]
LGTM, just one question about `test.yml`

File: .github/workflows/test.yml (hunk 3 above)
Comment (bgiori) [reply .github/workflows/test.yml:3]: Thanks, I turned this off when I was testing to reduce noise 👍 

I'll do the same all all other PRs.

**Pull request merged**